pilot/pkg/simulation/traffic.go

```diff
@@ -427,7 +427,7 @@ func (sim *Simulation) matchFilterChain(chains []*listener.FilterChain, defaultC
 	}, func(fc *listener.FilterChainMatch) bool {
 		if !hasTLSInspector {
 			// Without tls inspector, transport protocol will always be raw buffer
-			return fc.GetTransportProtocol() == xdsfilters.RawBufferTransportProtocol
+			return fc.GetTransportProtocol() == xdsfilters.RawBufferTransportProtocol && input.TLS == Plaintext
```

Comment:
I believe if there is no TLS inspector, then plaintext and TLS and mTLS traffic will all become "raw_buffer". Since it doesn't sniff any bytes at all, envoy would have no way to tell if its tls or not - it sets everything to raw_buffer